src/resolve.ts

```diff
@@ -118,7 +118,7 @@ export function resolvePath (id: string, opts?: ResolveOptions) {
 }
 
 export function createResolve (defaults?: ResolveOptions) {
-  return (id, url) => {
+  return (id: string, url?: ResolveOptions['url']) => {
     return resolve(id, { url, ...defaults })
```

Comment:
I've also noticed that the logic here is incorrect as the defaults will override the `url` even when it is explicitly provided. I didn't want to mix that fix in into this one but I can create a follow-up PR for that.